compiler/lib/code.ml

```diff
@@ -787,12 +787,32 @@ let eq p1 p2 =
        (fun pc block1 b ->
          b
          &&
-         try
-           let block2 = Addr.Map.find pc p2.blocks in
-           Poly.(block1.params = block2.params)
-           && Poly.(block1.branch = block2.branch)
-           && Poly.(block1.body = block2.body)
-         with Not_found -> false)
+         match Addr.Map.find pc p2.blocks with
+         | exception Not_found -> false
+         | block2 ->
+             List.equal ~eq:Var.equal block1.params block2.params
+             && Poly.(block1.branch = block2.branch)
+             && List.equal
+                  ~eq:(fun i i' ->
+                    match i, i' with
+                    | Let (x, Constant (Float f)), Let (x', Constant (Float f')) ->
```

Comment:
- `constant` is recursive and may contain floats under `Tuple`.
- I was wondering whether we should use the bit representation  of float in the IR. to avoid such thing to happen again in the future.

Reply:
- Oh, indeed!
- I tried that. What to you think of it?

Reply:
I like the new state